src/skeleton.tsx

```diff
@@ -0,0 +1,49 @@
+import { keyframes } from '@emotion/react';
+import React from 'react';
+import { Box, BoxProps } from 'theme-ui';
+
+export interface SkeletonProps extends BoxProps {
+  /** `circular` | `rectangular` | `text` */
+  shape: 'circular' | 'rectangular' | 'text';
+
+  /**
+   * Use to set size of circular shape.
+   *
+   * @default 1.2rem
+   */
+  size?: string | number;
+}
+
+const pulseKeyframe = keyframes({
+  '0%': {
+    opacity: 1,
+  },
+  '50%': {
+    opacity: 0.4,
+  },
+  '100%': {
+    opacity: 1,
+  },
+});
+
+export const Skeleton = React.forwardRef<HTMLDivElement, SkeletonProps>(
+  ({ shape = 'text', size = '1.2em', ...props }, ref) => (
+    <Box
+      ref={ref}
+      variant="primary"
+      {...props}
+      // @ts-expect-error themeKey is a private prop
+      __themeKey="skeleton"
+      __css={{
+        backgroundColor: 'muted',
+        height: shape === 'circular' ? size : shape === 'text' && 'auto',
```

Comment:
```suggestion
				// Use `size` value on circular shape and auto if shape is text, otherwise undefined.
        height: shape === 'circular' ? size : shape === 'text' && 'auto',
```